server/src/main/java/org/elasticsearch/index/mapper/DynamicTemplate.java

```diff
@@ -60,6 +146,12 @@ public static MatchType fromString(String value) {
 
         /** Whether {@code value} matches {@code regex}. */
         public abstract boolean matches(String regex, String value);
+
+        /**
+         * Whether the {@code pattern} is a valid for the MatchType.
+         * @throws IllegalArgumentException if the pattern is not valid
+         */
+        public abstract void isValidPattern(String pattern);
```

Comment:
I think I'd call this `validate` - `isXXX` implies a boolean return value to me.

Reply:
Yeah, good call. The original version I did returned boolean, but I need the error message on failure. I'll change it.